### DISCUSSION TITLE
Welcome to CS50x Discussions!

### DISCUSSION THREAD

**Discussion by 2013xile** [Xilesun/CS50x#1] [Announcements]
## 👋 Welcome!

本讨论区旨在围绕[CS50x课程](https://cs50.harvard.edu/x/2023/)进行学习、讨论、交流。

提问前烦请阅读[How-To-Ask-Questions-The-Smart-Way](https://github.com/ryanhanwu/How-To-Ask-Questions-The-Smart-Way/blob/main/README-zh_CN.md)

提问步骤：

1. 进入 [https://github.com/Xilesun/CS50x/discussions](https://github.com/Xilesun/CS50x/discussions) - New discussion
<img width="1597" alt="image" src="https://github.com/Xilesun/CS50x/assets/3250534/e8aa52da-73ae-4c0d-bfc0-385eaac8215e">

2. Q&A - Get started
<img width="1560" alt="image" src="https://github.com/Xilesun/CS50x/assets/3250534/16a03b11-499d-4bb9-9c0f-6fc380ffb100">

3. 填写表单
<img width="1052" alt="image" src="https://github.com/Xilesun/CS50x/assets/3250534/9eff23ba-d81d-42ca-8baa-53733058a467">

**Comment by kitaku-D** [comment]
老哥，请问一下在WEEK0的视频中，您的scratch是是如何定义time这个变量的，他是以怎样的形式累计值的呢

**Reply by 2013xile** [reply]
你是指我的视频吗，我看了下我只定义了 `times` 这个变量，是指这个吗？
首先我是用 **set `times` to 1** 将它初始化为1，在后面 `repeat` 的时候，每次 **change `times` by 1**, 即每次循环给 `times` 往上加1.

**Reply by kitaku-D** [reply]
我明白了大佬！谢谢您的耐心指导！

**Comment by justwant121** [comment]
请问如果window系统不装linux，直接使用vscode与装了linux之后有什么区别呢？因为之前用的都是codebloks学c，所以有很多都是会用，但更底层的东西都不太清楚。

**Reply by 2013xile** [reply]
你好。VSCode本身的用法大多是没有区别的，Windows环境下主要就是一些东西的配置和用法不太一样，包括环境变量配置、命令行命令等。在前期，如果是比较熟悉Windows环境大多数情况下问题应该是不大的，主要是对于一些新手朋友，可能对系统本身不太熟悉，用Windows可能就不太好跟着教程和视频的步骤操作。在后面进行更深入开发的时候，虽然Windows现在有不少优化，但是根据大多数朋友的反馈，使用Linux的体验还是会更好，当然到那时候你也有自己的判断了。

**Comment by justwant121** [comment]
我明白了，非常感谢你的回答！



在 2023-11-12 20:00:54，"YANG QIA" ***@***.***> 写道：

你好。VSCode本身的用法大多是没有区别的，Windows环境下主要就是一些东西的配置和用法不太一样，包括环境变量配置、命令行命令等。在前期，如果是比较熟悉Windows环境大多数情况下问题应该是不大的，主要是对于一些新手朋友，可能对系统本身不太熟悉，用Windows可能就不太好跟着教程和视频的步骤操作。在后面进行更深入开发的时候，虽然Windows现在有不少优化，但是根据大多数朋友的反馈，使用Linux的体验还是会更好，当然到那时候你也有自己的判断了。

**Comment by linjiash** [comment]
大佬，我一直在终端报错怎么办？
正在启动生成...
/usr/bin/clang -fcolor-diagnostics -fansi-escape-codes -g /Users/wangqiong/test/hello.c -o /Users/wangqiong/test/hello
ld: Undefined symbols:
  _get_string, referenced from:
      _main in hello-3a6d02.o
clang: error: linker command failed with exit code 1 (use -v to see invocation)

生成已完成，但出现错误。

 *  终端进程启动失败(退出代码: -1)。 
 *  终端将被任务重用，按任意键关闭。

preLaunchTask“C/C++: clang 生成活动文件”已终止，退出代码为 -1。运行就报这个
然后跳转到json

**Reply by 2013xile** [reply]
麻烦描述一下你的操作过程，点了什么按钮，或者是按照哪里操作的。
你是什么系统 MacOS？本地使用的时候安装 `cs50.h` 库了吗？

**Reply by linjiash** [reply]
mac os本地也安装了cs50，按照老师第一节课的hello，answer一模一样照抄下去，然而还是失败了，是不是安装cs50的时候出现问题了，前面的hello world是可以运行的

**Reply by 2013xile** [reply]
环境变量也设置了？编译的时候加上 `-lcs50` 了吗？看一下这一节课呢 https://www.bilibili.com/video/BV1Lj41197eZ/

**Reply by linjiash** [reply]
刚刚看到环境变量那个课，解决了，谢谢
